### DISCUSSION TITLE
Mapping only non-null properties from source to target

### DISCUSSION THREAD

**Discussion by moxplod** [riok/mapperly#452] [Q&A]
Hi, new user of Mapperly. I love the simplicity and speed of usage. 

I have a Model class Client with all nullable properties (it is used in EF if that is relevant). And ClientDto with similar nullable properties. I use ClientDto to get data from a few sources, like import, webhooks, and our web UI, and that data is stored in Client model. And I have this mapper configured: 

```

[Mapper]
public partial class ClientMapper
{
    public partial ClientDto ClientToClientDto(Client client);
    public partial void ClientDtoToClient(ClientDto clientDto, Client client);
}

```
Everything works great, except in ClientDtoToClient I only want to overwrite the non-null values from the source ClientDto. So the non-null values that are already existing in the target Client are not overwritten by null. 

This use case is coming from our integrations and csv import, where we want the users to be able to update a single Property in ClientDto.

What would be the recommended way to do that?

**Comment by latonz** [comment]
This is not yet possible, improvements which allow this are tracked in https://github.com/riok/mapperly/issues/98. Feel free to give it a 👍 to increase its priority in our backlog.
